app/repository/TpsRepo.scala

```diff
@@ -36,4 +36,6 @@ final class TpsRepo @Inject() (reactiveMongoComponent: ReactiveMongoComponent, c
     )
   )
 
+  def findPayment(tpsId: TpsId): Future[Option[TpsPayments]] = findById(tpsId)
```

Comment:
You might want to add `getPaymentId(id): Future[P] = findPayment(id).map(_.getOrElse(throw....))` and use it whenever it's expected that the records existed.

Reply:
don't think we need it yet